recipes/capnproto/all/conanfile.py

```diff
@@ -0,0 +1,185 @@
+        "with_zlib": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+        "with_zlib": True
+    }
+
+    _cmake = None
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn("Cap'n Proto requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("Cap'n Proto requires C++14, which your compiler does not support.")
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("Cap'n Proto doesn't support shared libraries for Visual Studio")
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1h")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+
+    def build_requirements(self):
+        if self.settings.os != "Windows":
+            self.build_requires("libtool/2.4.6")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["EXTERNAL_CAPNP"] = False
+        self._cmake.definitions["CAPNP_LITE"] = False
+        self._cmake.definitions["WITH_OPENSSL"] = self.options.with_openssl
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        args = [
+            "--enable-shared" if self.options.shared else "--disable-shared",
+            "--disable-static" if self.options.shared else "--enable-static",
+            "--with-openssl" if self.options.with_openssl else "--without-openssl",
+            "--with-zlib" if self.options.with_zlib else "--without-zlib",
+            "--enable-reflection"
```

Comment:
Do autotools and cmake have the same *reflection* behavior?

Reply:
yes, reflection disabled is `CAPNP_LITE ON` in CMake